src/README.md

```diff
@@ -1,246 +1,21 @@
-
-Hiding KV-cache introduces a peculiarity for beam search algorithm. Beam search suggests batched inference of multiple beams. The design described here so far would result in generating multiple independent sequences of tokens. Beam search algorithm, on the other hand, requires removing some of the ongoing beams and splitting other beams to multiple branches. Beam removal requires deleting corresponding KV-cache entry and beam splitting requires copying corresponding KV-cache values.
-
-To provide the possibility to implement beam search without accessing model's internal state, a stateful LLM converted with `optimum-intel` or [llm_bench](../../../llm_bench/python/) introduces an additional 1-dimentional `beam_idx` input. `beam_idx` must contain indexes of elements in a batch which are intended to be selected and will evolve during the next beam search iteration. There's only one beam when the generation starts. That beam corresponds to the initial prompt. `beam_idx` must have values: `[0, 0]` to keep the initial beam and introduce its copy. The dynamic batch size enables to change the number of beams dynamically. `beam_idx` must have `[1]` as the value to remove zeroth sequence and keep the second beam only.
-
-Assume there are two running beams. To proceed with generating both beams at the next iteration, `beam_idx` values must be `[0, 1]`, pointing to batch elements `0` and `1`. To drop the last beam and split the other beam in two, `beam_idx` must be set to `[0, 0]`. This results in utilizing only the part of KV cache corresponding to the zeroth element in the batch. The process of selecting proper entries in cache is called Cache Reorder.
-
-![](beam_idx-fork.gif)
-![](beam_idx-drop.gif)
-
-The images below represent stateless and stateful LLM pipelines. The model has 4 inputs:
-1. `input_ids` contains the next selected token
-2. `attention_mask` is filled with `1`
-3. `position_ids` encodes a position of currently generating token in the sequence
-4. `beam_idx` selects beams
-
-The model has 1 output `logits` describing the predicted distribution over the next tokens. And there's KV cache state.
-
-![](stateless.jpg)
-![](stateful.jpg)
+For information on how OpenVINO™ GenAI works, refer to the [How It Works Section](./docs/HOW_IT_WORKS.md).
 
 ## Supported models
 
-1. chatglm
-   1. https://huggingface.co/THUDM/chatglm2-6b - refer to
-   [chatglm2-6b - AttributeError: can't set attribute](../../../llm_bench/python/doc/NOTES.md#chatglm2-6b---attributeerror-cant-set-attribute)
```

Comment:
Not sure if we need to keep troubleshooting hint links in the "Supported Models" table.
But if we need such links (now there are only for `chatglm2-6b` and `Qwen-7B-Chat-Int4` models) - I would suggest to place them in the end of `SUPPORTED_MODELS.md` file in "Troubleshooting" section.

Reply:
No need to have them. These are for llm_bench